Libraries/LibWeb/HTML/MessagePort.cpp

```diff
@@ -438,17 +438,33 @@ void MessagePort::close()
         disentangle();
 }
 
-#undef __ENUMERATE
-#define __ENUMERATE(attribute_name, event_name)                         \
-    void MessagePort::set_##attribute_name(WebIDL::CallbackType* value) \
-    {                                                                   \
-        set_event_handler_attribute(event_name, value);                 \
-    }                                                                   \
-    WebIDL::CallbackType* MessagePort::attribute_name()                 \
-    {                                                                   \
-        return event_handler_attribute(event_name);                     \
-    }
-ENUMERATE_MESSAGE_PORT_EVENT_HANDLERS(__ENUMERATE)
-#undef __ENUMERATE
+// https://html.spec.whatwg.org/multipage/web-messaging.html#handler-messageeventtarget-onmessageerror
+void MessagePort::set_onmessageerror(GC::Ptr<WebIDL::CallbackType> value)
+{
+    set_event_handler_attribute(EventNames::messageerror, value);
+}
+
+// https://html.spec.whatwg.org/multipage/web-messaging.html#handler-messageeventtarget-onmessageerror
+GC::Ptr<WebIDL::CallbackType> MessagePort::onmessageerror()
+{
+    return event_handler_attribute(EventNames::messageerror);
+}
+
+// https://html.spec.whatwg.org/multipage/web-messaging.html#handler-messageeventtarget-onmessage
+void MessagePort::set_onmessage(GC::Ptr<WebIDL::CallbackType> value)
+{
+    set_event_handler_attribute(EventNames::message, value);
+
+    // https://html.spec.whatwg.org/multipage/web-messaging.html#message-ports:handler-messageeventtarget-onmessage
+    // The first time a MessagePort object's onmessage IDL attribute is set, the port's port message queue must be enabled,
+    // as if the start() method had been called.
+    start();
```

Comment:
@ADKaster interesting tidbit from the spec i noticed

Reply:
I was thinking it shouldn't matter since [start()](https://html.spec.whatwg.org/multipage/web-messaging.html#dom-messageport-start) says:

> The start() method steps are to enable [this](https://webidl.spec.whatwg.org/#this)'s [port message queue](https://html.spec.whatwg.org/multipage/web-messaging.html#port-message-queue), if it is not already enabled.

But this made me think a little bit deeper, I suppose it could technically be different if spec means that you could set `onmessage` with implicit `start()`, follow it with a `close()`, and then a set of `onmessage` again is meant to do nothing. But this doesn't appear to be the case from testing.